pkg/flow/logging/logger.go

```diff
@@ -96,3 +103,42 @@ func (fw *lokiWriter) Write(p []byte) (int, error) {
 	}
 	return len(p), nil
 }
+
+type formatVar struct {
+	mut sync.RWMutex
+	f   Format
+}
+
+func (f *formatVar) Format() Format {
+	f.mut.RLock()
+	defer f.mut.RUnlock()
+	return f.f
+}
+
+func (f *formatVar) Set(format Format) {
+	f.mut.Lock()
+	defer f.mut.Unlock()
+	f.f = format
+}
+
+type writerVar struct {
+	mut sync.RWMutex
+	w   io.Writer
+}
+
+func (w *writerVar) Set(inner io.Writer) {
+	w.mut.Lock()
+	defer w.mut.Unlock()
+	w.w = inner
+}
+
+func (w *writerVar) Write(p []byte) (n int, err error) {
+	w.mut.RLock()
+	defer w.mut.RUnlock()
+
+	if w.w == nil {
+		return len(p), nil
```

Comment:
It matters a little; implementations of io.Writer are expected to either return `len(p)` or an error if they return less than `len(p)`. 

You can even see this behavior in the implementation of io.MultiWriter: https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/io/multi.go;l=113-116

So in this case, since we've fully consumed the write (we just didn't do anything with it), we should return `len(p)`. I'll explain this in a comment.

Reply:
Actually, maybe it should be an error. I'll start with that and we'll see what happens.